rust/src/schema.rs

```diff
@@ -97,54 +228,51 @@ impl Field {
             "double" => DataType::Float64,
             "binary" => DataType::Binary,
             "string" => DataType::Utf8,
-            _ => panic!(),
+            x => DataType::Extension(
```

Comment:
It should return error, instead of an silent error (this will be propogated within the system, and hard to debug)